disasm-test/Main.hs

```diff
@@ -528,6 +534,104 @@ runAssemblyTest llvmVersion knownBugs sweet expct
                        Nothing -> error "Failed processLL"
 
 
+
+-- | Compare two ASTs to see if they are the same.  Fundamentally is just done
+--  via (==) on the normalized ASTs, but this first uses the tree-diff package to
+--  generate an nicer output to allow focus on the actual diffs rather than
+--  leaving it to the user to analyze the large blobs to find out where.
+cmpASTs :: AST.Module -> AST.Module -> TestM ()
+cmpASTs ast1 ast2 = do
+  let d = ediff ast1 ast2
+      msg = "Differences (marked with + and - line prefixes:\n"
+            <> show (prettyEditExprCompact d)
+  liftIO $ assertBool msg $ ast1 == ast2
+
+deriving instance ToExpr (AST.Type' AST.Ident)
```

Comment:
It might be worth moving these orphan instances to their own module (akin to `unit-test/Tests/ExpressionInstances.hs`) to reduce the time it takes to rebuild `disasm-test/Main.hs` on its own.